Issue on NULL DNS values -  Proxmox VE 8.1.4

**Describe the bug**
When deploy a virtual machine with dhcp for interface address, the plugin crashes

**To Reproduce**
Steps to reproduce the behavior:

1. Create a sample deploy_vm and set ip address as dhcp
2. terraform apply
3. See error

Please also provide a minimal Terraform configuration that reproduces the issue.

```hcl
 resource "proxmox_virtual_environment_vm" "ubuntu_vm" {
  name        = var.vm_name
  description = var.vm_description
  tags        = var.vm_tags
  node_name = var.proxmox_node_name

  agent {
    enabled = true
  }

  startup {
    order      = var.vm_startup_order
    up_delay   = var.vm_up_delay
    down_delay = var.vm_down_delay
  }

  disk {
    size = var.vm_disk_size_gb
    file_id      = var.vm_source_image_path    
    interface    = var.vm_disk_interface    
    datastore_id = var.vm_disk_datastore_id
  }

  cpu {
    cores =  var.vm_cpu_cores
    sockets = var.vm_cpu_sockets
  }
  
  memory {
    dedicated = var.vm_memory_dedicated
    shared = var.vm_memory_shared
    floating = var.vm_memory_floating
  }

  initialization {
    datastore_id = var.vm_disk_cloudinit_datastore_id

    ip_config {
      ipv4 {
        address = "dhcp"
      }
    }
    
    dns {
      servers = var.vm_ip_dns
      domain = var.vm_ip_domain
    }

    user_data_file_id = proxmox_virtual_environment_file.cloud_config.id
  }

  network_device {
    bridge = var.vm_network
  }

  operating_system {
    type = var.vm_os_type
  }

  serial_device {}
}
--SNIP--
```

**Expected behavior**
Vm Created without issues

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

- Single or clustered Proxmox: single
- Provider version (ideally it should be the latest version): 0.43.3
- Terraform version: Terraform v1.6.3
- OS (where you run Terraform from): linux ubuntu x86_64
- Debug logs (`TF_LOG=DEBUG terraform apply`):
```
leonardo@Dell-LeonFelix:/mnt/d/IaC/fivenet/proxmox_rancher$ TF_LOG=DEBUG terraform apply --var-file fivenet.tfvars 
--SNIP--
Terraform will perform the following actions:

  # proxmox_virtual_environment_vm.ubuntu_vm will be created
  + resource "proxmox_virtual_environment_vm" "ubuntu_vm" {
      + acpi                    = true
      + bios                    = "seabios"
      + description             = "Managed by Terraform"
      + id                      = (known after apply)
      + ipv4_addresses          = (known after apply)
      + ipv6_addresses          = (known after apply)
      + keyboard_layout         = "en-us"
      + mac_addresses           = (known after apply)
      + migrate                 = false
      + name                    = "test-deploy-terraform"
      + network_interface_names = (known after apply)
      + node_name               = "pve-fnet"
      + on_boot                 = true
      + reboot                  = false
      + scsi_hardware           = "virtio-scsi-pci"
      + started                 = true
      + stop_on_destroy         = false
      + tablet_device           = true
      + tags                    = [
          + "terraform",
          + "ubuntu",
        ]
      + template                = false
      + timeout_clone           = 1800
      + timeout_create          = 1800
      + timeout_migrate         = 1800
      + timeout_move_disk       = 1800
      + timeout_reboot          = 1800
      + timeout_shutdown_vm     = 1800
      + timeout_start_vm        = 1800
      + timeout_stop_vm         = 300
      + vm_id                   = (known after apply)

      + agent {
          + enabled = true
          + timeout = "15m"
          + trim    = false
          + type    = "virtio"
        }

      + cpu {
          + architecture = "x86_64"
          + cores        = 2
          + hotplugged   = 0
          + limit        = 0
          + numa         = false
          + sockets      = 2
          + type         = "qemu64"
          + units        = 1024
        }

      + disk {
          + cache             = "none"
          + datastore_id      = "local"
          + discard           = "ignore"
          + file_format       = (known after apply)
          + file_id           = "local:iso/jammy-server-cloudimg-amd64.img"
          + interface         = "scsi0"
          + iothread          = false
          + path_in_datastore = (known after apply)
          + size              = 300
          + ssd               = false
        }

      + initialization {
          + datastore_id      = "local"
          + user_data_file_id = "local:snippets/test-deploy-terraform.cloud-config.yaml"

          + dns {}

          + ip_config {
              + ipv4 {
                  + address = "dhcp"
                }
            }
        }

      + memory {
          + dedicated = 2048
          + floating  = 0
          + shared    = 0
        }

      + network_device {
          + bridge      = "vmbr1"
          + enabled     = true
          + firewall    = false
          + mac_address = (known after apply)
          + model       = "virtio"
          + mtu         = 0
          + queues      = 0
          + rate_limit  = 0
          + vlan_id     = 0
        }

      + operating_system {
          + type = "l26"
        }

      + serial_device {
          + device = "socket"
        }

      + startup {
          + down_delay = 30
          + order      = 3
          + up_delay   = 30
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + ubuntu_vm_address     = (known after apply)
2024-01-16T15:15:18.793+0100 [DEBUG] command: asking for input: "\nDo you want to perform these actions?"

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

2024-01-16T15:15:20.049+0100 [INFO]  backend/local: apply calling Apply
2024-01-16T15:15:20.049+0100 [DEBUG] Building and walking apply graph for NormalMode plan
2024-01-16T15:15:20.049+0100 [DEBUG] Resource state not found for node "proxmox_virtual_environment_vm.ubuntu_vm", instance proxmox_virtual_environment_vm.ubuntu_vm
2024-01-16T15:15:20.049+0100 [DEBUG] adding implicit provider configuration provider["registry.terraform.io/hashicorp/random"], implied first by random_password.ubuntu_vm_password (expand)
2024-01-16T15:15:20.049+0100 [DEBUG] adding implicit provider configuration provider["registry.terraform.io/hashicorp/tls"], implied first by tls_private_key.ubuntu_vm_key (expand)
2024-01-16T15:15:20.050+0100 [DEBUG] ProviderTransformer: "proxmox_virtual_environment_vm.ubuntu_vm" (*terraform.NodeApplyableResourceInstance) needs provider["registry.terraform.io/bpg/proxmox"]
2024-01-16T15:15:20.050+0100 [DEBUG] ProviderTransformer: "proxmox_virtual_environment_vm.ubuntu_vm (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/bpg/proxmox"]
2024-01-16T15:15:20.050+0100 [DEBUG] ProviderTransformer: "random_password.ubuntu_vm_password (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/hashicorp/random"]
2024-01-16T15:15:20.050+0100 [DEBUG] ProviderTransformer: "tls_private_key.ubuntu_vm_key (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/hashicorp/tls"]
2024-01-16T15:15:20.050+0100 [DEBUG] ProviderTransformer: "proxmox_virtual_environment_file.cloud_config (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/bpg/proxmox"]       
2024-01-16T15:15:20.050+0100 [DEBUG] ReferenceTransformer: "var.vm_tags" references: []
2024-01-16T15:15:20.050+0100 [DEBUG] ReferenceTransformer: "var.proxmox_node_name" references: []
2024-01-16T15:15:20.050+0100 [DEBUG] ReferenceTransformer: "var.vm_startup_order" references: []
2024-01-16T15:15:20.050+0100 [DEBUG] ReferenceTransformer: "var.vm_source_image_path" references: []
2024-01-16T15:15:20.050+0100 [DEBUG] ReferenceTransformer: "var.vm_down_delay" references: []
2024-01-16T15:15:20.050+0100 [DEBUG] ReferenceTransformer: "proxmox_virtual_environment_vm.ubuntu_vm (expand)" references: [var.vm_description var.vm_tags var.proxmox_node_name var.vm_name var.vm_down_delay var.vm_startup_order var.vm_up_delay var.vm_disk_size_gb var.vm_source_image_path var.vm_disk_interface var.vm_disk_datastore_id var.vm_cpu_cores var.vm_cpu_sockets var.vm_memory_dedicated var.vm_memory_floating var.vm_memory_shared var.vm_disk_cloudinit_datastore_id proxmox_virtual_environment_file.cloud_config (expand) var.vm_ip_gateway var.vm_ip_address var.vm_ip_dns var.vm_ip_domain var.vm_network var.vm_os_type]
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "random_password.ubuntu_vm_password (expand)" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.proxmox_node_ip_address" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_name" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_ip_address" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "output.ubuntu_vm_address (expand)" references: [proxmox_virtual_environment_vm.ubuntu_vm (expand) proxmox_virtual_environment_vm.ubuntu_vm proxmox_virtual_environment_vm.ubuntu_vm]
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/bpg/proxmox\"]" references: [var.proxmox_endpoint_address var.proxmox_insecure var.proxmox_password var.proxmox_temp_dir var.proxmox_username var.proxmox_use_ssh_agent var.proxmox_ssh_username var.proxmox_node_ip_address var.proxmox_node_name]
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "tls_private_key.ubuntu_vm_key (expand)" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_cpu_sockets" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_memory_dedicated" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_up_delay" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_disk_interface" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_memory_floating" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_description" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_disk_cloudinit_datastore_id" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_cpu_cores" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "output.ubuntu_vm_public_key (expand)" references: [tls_private_key.ubuntu_vm_key (expand)]
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/hashicorp/tls\"]" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "proxmox_virtual_environment_file.cloud_config (expand)" references: [var.proxmox_node_name var.vm_disk_cloudinit_datastore_id random_password.ubuntu_vm_password (expand) var.vm_name tls_private_key.ubuntu_vm_key (expand) var.vm_ssh_auth_key var.vm_ssh_auth_key var.vm_name]
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.proxmox_username" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_disk_datastore_id" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_ip_domain" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_disk_size_gb" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_memory_shared" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.proxmox_endpoint_address" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.proxmox_ssh_username" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.proxmox_insecure" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "output.ubuntu_vm_private_key (expand)" references: [tls_private_key.ubuntu_vm_key (expand)]
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/hashicorp/random\"]" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_ssh_auth_key" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_network" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "output.ubuntu_vm_password (expand)" references: [random_password.ubuntu_vm_password (expand)]
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "proxmox_virtual_environment_vm.ubuntu_vm" references: [var.vm_description var.vm_tags var.proxmox_node_name var.vm_name var.vm_up_delay var.vm_down_delay var.vm_startup_order var.vm_source_image_path var.vm_disk_interface var.vm_disk_size_gb var.vm_disk_datastore_id var.vm_cpu_cores var.vm_cpu_sockets var.vm_memory_dedicated var.vm_memory_floating var.vm_memory_shared var.vm_disk_cloudinit_datastore_id proxmox_virtual_environment_file.cloud_config (expand) var.vm_ip_gateway var.vm_ip_address var.vm_ip_domain var.vm_ip_dns var.vm_network var.vm_os_type]
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.proxmox_use_ssh_agent" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.proxmox_password" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_os_type" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_ip_gateway" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.vm_ip_dns" references: []
2024-01-16T15:15:20.051+0100 [DEBUG] ReferenceTransformer: "var.proxmox_temp_dir" references: []
2024-01-16T15:15:20.052+0100 [DEBUG] Starting graph walk: walkApply
2024-01-16T15:15:20.057+0100 [DEBUG] created provider logger: level=debug
2024-01-16T15:15:20.057+0100 [INFO]  provider: configuring client automatic mTLS
2024-01-16T15:15:20.062+0100 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/random/3.6.0/linux_amd64/terraform-provider-random_v3.6.0_x5 args=[".terraform/providers/registry.terraform.io/hashicorp/random/3.6.0/linux_amd64/terraform-provider-random_v3.6.0_x5"]
2024-01-16T15:15:20.067+0100 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/random/3.6.0/linux_amd64/terraform-provider-random_v3.6.0_x5 pid=2182
2024-01-16T15:15:20.068+0100 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/random/3.6.0/linux_amd64/terraform-provider-random_v3.6.0_x5
2024-01-16T15:15:20.284+0100 [INFO]  provider.terraform-provider-random_v3.6.0_x5: configuring server automatic mTLS: timestamp="2024-01-16T15:15:20.284+0100"
2024-01-16T15:15:20.305+0100 [DEBUG] provider: using plugin: version=5
2024-01-16T15:15:20.305+0100 [DEBUG] provider.terraform-provider-random_v3.6.0_x5: plugin address: address=/tmp/plugin494499483 network=unix timestamp="2024-01-16T15:15:20.305+0100"
2024-01-16T15:15:20.330+0100 [DEBUG] created provider logger: level=debug
2024-01-16T15:15:20.330+0100 [INFO]  provider: configuring client automatic mTLS
2024-01-16T15:15:20.334+0100 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/tls/4.0.5/linux_amd64/terraform-provider-tls_v4.0.5_x5 args=[".terraform/providers/registry.terraform.io/hashicorp/tls/4.0.5/linux_amd64/terraform-provider-tls_v4.0.5_x5"]
2024-01-16T15:15:20.339+0100 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/tls/4.0.5/linux_amd64/terraform-provider-tls_v4.0.5_x5 pid=2191
2024-01-16T15:15:20.339+0100 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/tls/4.0.5/linux_amd64/terraform-provider-tls_v4.0.5_x5
2024-01-16T15:15:20.602+0100 [INFO]  provider.terraform-provider-tls_v4.0.5_x5: configuring server automatic mTLS: timestamp="2024-01-16T15:15:20.602+0100"
2024-01-16T15:15:20.625+0100 [DEBUG] provider.terraform-provider-tls_v4.0.5_x5: plugin address: address=/tmp/plugin1720059456 network=unix timestamp="2024-01-16T15:15:20.625+0100"
2024-01-16T15:15:20.625+0100 [DEBUG] provider: using plugin: version=5
2024-01-16T15:15:20.649+0100 [DEBUG] created provider logger: level=debug
2024-01-16T15:15:20.649+0100 [INFO]  provider: configuring client automatic mTLS
2024-01-16T15:15:20.654+0100 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/bpg/proxmox/0.43.3/linux_amd64/terraform-provider-proxmox_v0.43.3 args=[".terraform/providers/registry.terraform.io/bpg/proxmox/0.43.3/linux_amd64/terraform-provider-proxmox_v0.43.3"]
2024-01-16T15:15:20.659+0100 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/bpg/proxmox/0.43.3/linux_amd64/terraform-provider-proxmox_v0.43.3 pid=2201
2024-01-16T15:15:20.659+0100 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/bpg/proxmox/0.43.3/linux_amd64/terraform-provider-proxmox_v0.43.3
2024-01-16T15:15:20.965+0100 [INFO]  provider.terraform-provider-proxmox_v0.43.3: configuring server automatic mTLS: timestamp="2024-01-16T15:15:20.964+0100"
2024-01-16T15:15:20.985+0100 [DEBUG] provider: using plugin: version=6
2024-01-16T15:15:20.985+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: plugin address: address=/tmp/plugin3394131311 network=unix timestamp="2024-01-16T15:15:20.985+0100"
2024-01-16T15:15:21.011+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-01-16T15:15:21.013+0100 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/random/3.6.0/linux_amd64/terraform-provider-random_v3.6.0_x5 pid=2182
2024-01-16T15:15:21.013+0100 [DEBUG] provider: plugin exited
2024-01-16T15:15:21.014+0100 [DEBUG] provider.terraform-provider-tls_v4.0.5_x5: Configuring provider: @caller=github.com/hashicorp/terraform-provider-tls/internal/provider/provider.go:107 @module=tls tf_provider_addr=registry.terraform.io/hashicorp/tls tf_req_id=b94f9791-ddb8-bedf-237c-25c5e20cb3b8 tf_rpc=Configure timestamp="2024-01-16T15:15:21.014+0100"
2024-01-16T15:15:21.014+0100 [DEBUG] provider.terraform-provider-tls_v4.0.5_x5: No proxy configuration detected: using provider defaults: tf_provider_addr=registry.terraform.io/hashicorp/tls tf_req_id=b94f9791-ddb8-bedf-237c-25c5e20cb3b8 tf_rpc=Configure @caller=github.com/hashicorp/terraform-provider-tls/internal/provider/provider.go:124 @module=tls timestamp="2024-01-16T15:15:21.014+0100"
2024-01-16T15:15:21.015+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-01-16T15:15:21.016+0100 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/tls/4.0.5/linux_amd64/terraform-provider-tls_v4.0.5_x5 pid=2191
2024-01-16T15:15:21.016+0100 [DEBUG] provider: plugin exited
2024-01-16T15:15:21.053+0100 [INFO]  provider.terraform-provider-proxmox_v0.43.3: Configuring the Proxmox provider...: @caller=github.com/bpg/terraform-provider-proxmox/fwprovider/provider.go:196 @module=proxmox tf_req_id=48ccca27-35de-299e-b697-5a41914748a5 tf_mux_provider="*proto6server.Server" tf_provider_addr=registry.terraform.io/bpg/proxmox tf_rpc=ConfigureProvider timestamp="2024-01-16T15:15:21.051+0100"
2024-01-16T15:15:21.067+0100 [WARN]  provider.terraform-provider-proxmox_v0.43.3: unable to require attribute replacement: @module=sdk.helper_schema tf_attribute_path=vm_id tf_req_id=33aa92e0-1562-a5d4-b67b-b95661f92769 tf_rpc=PlanResourceChange @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/customdiff/force_new.go:32 error="ForceNew: No changes for vm_id" tf_mux_provider=tf5to6server.v5tov6Server tf_provider_addr=registry.terraform.io/bpg/proxmox tf_resource_type=proxmox_virtual_environment_vm timestamp="2024-01-16T15:15:21.067+0100"
2024-01-16T15:15:21.073+0100 [WARN]  Provider "registry.terraform.io/bpg/proxmox" produced an invalid plan for proxmox_virtual_environment_vm.ubuntu_vm, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .timeout_reboot: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .timeout_start_vm: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .acpi: planned value cty.True for a non-computed attribute
      - .on_boot: planned value cty.True for a non-computed attribute
      - .migrate: planned value cty.False for a non-computed attribute
      - .tablet_device: planned value cty.True for a non-computed attribute
      - .timeout_move_disk: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .timeout_create: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .timeout_migrate: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .timeout_shutdown_vm: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .reboot: planned value cty.False for a non-computed attribute
      - .started: planned value cty.True for a non-computed attribute
      - .keyboard_layout: planned value cty.StringVal("en-us") for a non-computed attribute
      - .timeout_clone: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .timeout_stop_vm: planned value cty.NumberIntVal(300) for a non-computed attribute
      - .bios: planned value cty.StringVal("seabios") for a non-computed attribute
      - .scsi_hardware: planned value cty.StringVal("virtio-scsi-pci") for a non-computed attribute
      - .template: planned value cty.False for a non-computed attribute
      - .stop_on_destroy: planned value cty.False for a non-computed attribute
      - .disk[0].iothread: planned value cty.False for a non-computed attribute
      - .disk[0].ssd: planned value cty.False for a non-computed attribute
      - .disk[0].cache: planned value cty.StringVal("none") for a non-computed attribute
      - .disk[0].discard: planned value cty.StringVal("ignore") for a non-computed attribute
      - .cpu[0].type: planned value cty.StringVal("qemu64") for a non-computed attribute
      - .cpu[0].limit: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .cpu[0].numa: planned value cty.False for a non-computed attribute
      - .cpu[0].hotplugged: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .cpu[0].units: planned value cty.NumberIntVal(1024) for a non-computed attribute
      - .cpu[0].architecture: planned value cty.StringVal("x86_64") for a non-computed attribute
      - .serial_device[0].device: planned value cty.StringVal("socket") for a non-computed attribute
      - .agent[0].timeout: planned value cty.StringVal("15m") for a non-computed attribute
      - .agent[0].trim: planned value cty.False for a non-computed attribute
      - .agent[0].type: planned value cty.StringVal("virtio") for a non-computed attribute
      - .network_device[0].enabled: planned value cty.True for a non-computed attribute
      - .network_device[0].mtu: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .network_device[0].queues: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .network_device[0].rate_limit: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .network_device[0].firewall: planned value cty.False for a non-computed attribute
      - .network_device[0].model: planned value cty.StringVal("virtio") for a non-computed attribute
      - .network_device[0].vlan_id: planned value cty.NumberIntVal(0) for a non-computed attribute
proxmox_virtual_environment_vm.ubuntu_vm: Creating...
2024-01-16T15:15:21.073+0100 [INFO]  Starting apply for proxmox_virtual_environment_vm.ubuntu_vm
2024-01-16T15:15:21.074+0100 [DEBUG] proxmox_virtual_environment_vm.ubuntu_vm: applying the planned Create change
2024-01-16T15:15:21.076+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: 2024/01/16 15:15:21 [DEBUG] setting computed for "ipv4_addresses" from ComputedKeys
2024-01-16T15:15:21.076+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: 2024/01/16 15:15:21 [DEBUG] setting computed for "network_interface_names" from ComputedKeys
2024-01-16T15:15:21.076+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: 2024/01/16 15:15:21 [DEBUG] setting computed for "ipv6_addresses" from ComputedKeys
2024-01-16T15:15:21.076+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: 2024/01/16 15:15:21 [DEBUG] setting computed for "mac_addresses" from ComputedKeys
**2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: panic: interface conversion: interface {} is nil, not map[string]interface {}**
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: goroutine 34 [running]:
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmGetCloudInitConfig(0xdc1aa0?)
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3:       github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:3106 +0x1274
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmCreateCustom({0x106d7f0, 0xc000449b20}, 0x168bf40?, {0xe38340?, 0xc00017d4d0?})
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3:       github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:2602 +0x12d3
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmCreate({0x106d7f0, 0xc000449b20}, 0x0?, {0xe38340, 0xc00017d4d0})    
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3:       github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:1688 +0x94
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0002f5260, {0x106d748, 0xc00017d1d0}, 0xd?, {0xe38340, 0xc00017d4d0})
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3:       github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:778 +0x11b
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0002f5260, {0x106d748, 0xc00017d1d0}, 0xc000565930, 0xc00004ad00, {0xe38340, 0xc00017d4d0})
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3:       github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:909 +0xa89
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0000113f8, {0x106d748?, 0xc00017d020?}, 0xc0002343c0)
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3:       github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/grpc_provider.go:1074 +0xdbc
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange({{0x1074e20?, 0xc0000113f8?}}, {0x106d748, 0xc00017d020}, 0x0?)
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3:       github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf5to6server/tf5to6server.go:37 +0x54
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0x106d780?, {0x106d748?, 0xc00017cd20?}, 0xc000234370)
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3:       github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x193
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc000273900, {0x106d748?, 0xc00017c2d0?}, 0xc000448000)
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3:       github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/tf6server/server.go:857 +0x56b
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0xeb7a40?, 0xc000273900}, {0x106d748, 0xc00017c2d0}, 0xc00004a180, 0x0)
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3:       github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:503 +0x169
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000365a0, {0x106d748, 0xc00017c240}, {0x10733d0, 0xc0004069c0}, 0xc000794120, 0xc0003a8c30, 0x167ed98, 0x0)
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3:       google.golang.org/grpc@v1.60.0/server.go:1372 +0xe03
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: google.golang.org/grpc.(*Server).handleStream(0xc0000365a0, {0x10733d0, 0xc0004069c0}, 0xc000794120)
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3:       google.golang.org/grpc@v1.60.0/server.go:1783 +0xfec
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: google.golang.org/grpc.(*Server).serveStreams.func2.1()
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3:       google.golang.org/grpc@v1.60.0/server.go:1016 +0x59
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3: created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 23
2024-01-16T15:15:21.082+0100 [DEBUG] provider.terraform-provider-proxmox_v0.43.3:       google.golang.org/grpc@v1.60.0/server.go:1027 +0x115
2024-01-16T15:15:21.083+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-01-16T15:15:21.083+0100 [ERROR] plugin6.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-01-16T15:15:21.085+0100 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/bpg/proxmox/0.43.3/linux_amd64/terraform-provider-proxmox_v0.43.3 pid=2201 error="exit status 2"
2024-01-16T15:15:21.086+0100 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2024-01-16T15:15:21.086+0100 [ERROR] vertex "proxmox_virtual_environment_vm.ubuntu_vm" error: Plugin did not respond
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-proxmox_v0.43.3 plugin:

**panic: interface conversion: interface {} is nil, not map[string]interface {}**

goroutine 34 [running]:
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmGetCloudInitConfig(0xdc1aa0?)
        github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:3106 +0x1274
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmCreateCustom({0x106d7f0, 0xc000449b20}, 0x168bf40?, {0xe38340?, 0xc00017d4d0?})
        github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:2602 +0x12d3
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmCreate({0x106d7f0, 0xc000449b20}, 0x0?, {0xe38340, 0xc00017d4d0})
        github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:1688 +0x94
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0002f5260, {0x106d748, 0xc00017d1d0}, 0xd?, {0xe38340, 0xc00017d4d0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:778 +0x11b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0002f5260, {0x106d748, 0xc00017d1d0}, 0xc000565930, 0xc00004ad00, {0xe38340, 0xc00017d4d0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:909 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0000113f8, {0x106d748?, 0xc00017d020?}, 0xc0002343c0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/grpc_provider.go:1074 +0xdbc
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange({{0x1074e20?, 0xc0000113f8?}}, {0x106d748, 0xc00017d020}, 0x0?)
        github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf5to6server/tf5to6server.go:37 +0x54
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0x106d780?, {0x106d748?, 0xc00017cd20?}, 0xc000234370)
        github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc000273900, {0x106d748?, 0xc00017c2d0?}, 0xc000448000)
        github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/tf6server/server.go:857 +0x56b
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0xeb7a40?, 0xc000273900}, {0x106d748, 0xc00017c2d0}, 0xc00004a180, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:503 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000365a0, {0x106d748, 0xc00017c240}, {0x10733d0, 0xc0004069c0}, 0xc000794120, 0xc0003a8c30, 0x167ed98, 0x0)
        google.golang.org/grpc@v1.60.0/server.go:1372 +0xe03
google.golang.org/grpc.(*Server).handleStream(0xc0000365a0, {0x10733d0, 0xc0004069c0}, 0xc000794120)
        google.golang.org/grpc@v1.60.0/server.go:1783 +0xfec
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.60.0/server.go:1016 +0x59
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 23
        google.golang.org/grpc@v1.60.0/server.go:1027 +0x115

Error: The terraform-provider-proxmox_v0.43.3 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

2024-01-16T15:15:21.092+0100 [DEBUG] provider: plugin exited
```

--- a/proxmoxtf/resource/vm.go
+++ b/proxmoxtf/resource/vm.go
@@ -3092,150 +3092,153 @@ func vmGetAudioDriverValidator() schema.SchemaValidateDiagFunc {
 	}, false))
 }
 
+//nolint:lll
 func vmGetCloudInitConfig(d *schema.ResourceData) *vms.CustomCloudInitConfig {
-	var initializationConfig *vms.CustomCloudInitConfig
-
 	initialization := d.Get(mkResourceVirtualEnvironmentVMInitialization).([]interface{})
 
-	if len(initialization) > 0 {
-		initializationBlock := initialization[0].(map[string]interface{})
-		initializationConfig = &vms.CustomCloudInitConfig{}
-		initializationDNS := initializationBlock[mkResourceVirtualEnvironmentVMInitializationDNS].([]interface{})
+	if len(initialization) == 0 || initialization[0] == nil {
+		return nil
+	}
 
-		if len(initializationDNS) > 0 {
-			initializationDNSBlock := initializationDNS[0].(map[string]interface{})
-			domain := initializationDNSBlock[mkResourceVirtualEnvironmentVMInitializationDNSDomain].(string)
+	var initializationConfig *vms.CustomCloudInitConfig
 
-			if domain != "" {
-				initializationConfig.SearchDomain = &domain
+	initializationBlock := initialization[0].(map[string]interface{})
+	initializationConfig = &vms.CustomCloudInitConfig{}
+	initializationDNS := initializationBlock[mkResourceVirtualEnvironmentVMInitializationDNS].([]interface{})
+
+	if len(initializationDNS) > 0 && initializationDNS[0] != nil {
+		initializationDNSBlock := initializationDNS[0].(map[string]interface{})
+		domain := initializationDNSBlock[mkResourceVirtualEnvironmentVMInitializationDNSDomain].(string)
+
+		if domain != "" {
+			initializationConfig.SearchDomain = &domain
+		}
+
+		servers := initializationDNSBlock[mkResourceVirtualEnvironmentVMInitializationDNSServers].([]interface{})
+		deprecatedServer := initializationDNSBlock[mkResourceVirtualEnvironmentVMInitializationDNSServer].(string)
+
+		if len(servers) > 0 {
+			nameserver := strings.Join(ConvertToStringSlice(servers), " ")
+
+			initializationConfig.Nameserver = &nameserver
+		} else if deprecatedServer != "" {
+			initializationConfig.Nameserver = &deprecatedServer
+		}
+	}
+
+	initializationIPConfig := initializationBlock[mkResourceVirtualEnvironmentVMInitializationIPConfig].([]interface{})
+	initializationConfig.IPConfig = make(
+		[]vms.CustomCloudInitIPConfig,
+		len(initializationIPConfig),
+	)
+
+	for i, c := range initializationIPConfig {
+		configBlock := c.(map[string]interface{})
+		ipv4 := configBlock[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv4].([]interface{})
+
+		if len(ipv4) > 0 {
+			ipv4Block := ipv4[0].(map[string]interface{})
+			ipv4Address := ipv4Block[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv4Address].(string)
+
+			if ipv4Address != "" {
+				initializationConfig.IPConfig[i].IPv4 = &ipv4Address
 			}
 
-			servers := initializationDNSBlock[mkResourceVirtualEnvironmentVMInitializationDNSServers].([]interface{})
-			deprecatedServer := initializationDNSBlock[mkResourceVirtualEnvironmentVMInitializationDNSServer].(string)
+			ipv4Gateway := ipv4Block[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv4Gateway].(string)
 
-			if len(servers) > 0 {
-				nameserver := strings.Join(ConvertToStringSlice(servers), " ")
-
-				initializationConfig.Nameserver = &nameserver
-			} else if deprecatedServer != "" {
-				initializationConfig.Nameserver = &deprecatedServer
+			if ipv4Gateway != "" {
+				initializationConfig.IPConfig[i].GatewayIPv4 = &ipv4Gateway
 			}
 		}
 
-		initializationIPConfig := initializationBlock[mkResourceVirtualEnvironmentVMInitializationIPConfig].([]interface{})
-		initializationConfig.IPConfig = make(
-			[]vms.CustomCloudInitIPConfig,
-			len(initializationIPConfig),
-		)
+		ipv6 := configBlock[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv6].([]interface{})
 
-		for i, c := range initializationIPConfig {
-			configBlock := c.(map[string]interface{})
-			ipv4 := configBlock[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv4].([]interface{})
+		if len(ipv6) > 0 {
+			ipv6Block := ipv6[0].(map[string]interface{})
+			ipv6Address := ipv6Block[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv6Address].(string)
 
-			if len(ipv4) > 0 {
-				ipv4Block := ipv4[0].(map[string]interface{})
-				ipv4Address := ipv4Block[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv4Address].(string)
-
-				if ipv4Address != "" {
-					initializationConfig.IPConfig[i].IPv4 = &ipv4Address
-				}
-
-				ipv4Gateway := ipv4Block[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv4Gateway].(string)
-
-				if ipv4Gateway != "" {
-					initializationConfig.IPConfig[i].GatewayIPv4 = &ipv4Gateway
-				}
+			if ipv6Address != "" {
+				initializationConfig.IPConfig[i].IPv6 = &ipv6Address
 			}
 
-			ipv6 := configBlock[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv6].([]interface{})
+			ipv6Gateway := ipv6Block[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv6Gateway].(string)
 
-			if len(ipv6) > 0 {
-				ipv6Block := ipv6[0].(map[string]interface{})
-				ipv6Address := ipv6Block[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv6Address].(string)
-
-				if ipv6Address != "" {
-					initializationConfig.IPConfig[i].IPv6 = &ipv6Address
-				}
-
-				ipv6Gateway := ipv6Block[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv6Gateway].(string)
-
-				if ipv6Gateway != "" {
-					initializationConfig.IPConfig[i].GatewayIPv6 = &ipv6Gateway
-				}
+			if ipv6Gateway != "" {
+				initializationConfig.IPConfig[i].GatewayIPv6 = &ipv6Gateway
 			}
 		}
+	}
 
-		initializationUserAccount := initializationBlock[mkResourceVirtualEnvironmentVMInitializationUserAccount].([]interface{})
+	initializationUserAccount := initializationBlock[mkResourceVirtualEnvironmentVMInitializationUserAccount].([]interface{})
 
-		if len(initializationUserAccount) > 0 {
-			initializationUserAccountBlock := initializationUserAccount[0].(map[string]interface{})
-			keys := initializationUserAccountBlock[mkResourceVirtualEnvironmentVMInitializationUserAccountKeys].([]interface{})
+	if len(initializationUserAccount) > 0 {
+		initializationUserAccountBlock := initializationUserAccount[0].(map[string]interface{})
+		keys := initializationUserAccountBlock[mkResourceVirtualEnvironmentVMInitializationUserAccountKeys].([]interface{})
 
-			if len(keys) > 0 {
-				sshKeys := make(vms.CustomCloudInitSSHKeys, len(keys))
+		if len(keys) > 0 {
+			sshKeys := make(vms.CustomCloudInitSSHKeys, len(keys))
 
-				for i, k := range keys {
-					sshKeys[i] = k.(string)
-				}
-
-				initializationConfig.SSHKeys = &sshKeys
+			for i, k := range keys {
+				sshKeys[i] = k.(string)
 			}
 
-			password := initializationUserAccountBlock[mkResourceVirtualEnvironmentVMInitializationUserAccountPassword].(string)
-
-			if password != "" {
-				initializationConfig.Password = &password
-			}
-
-			username := initializationUserAccountBlock[mkResourceVirtualEnvironmentVMInitializationUserAccountUsername].(string)
-
-			initializationConfig.Username = &username
+			initializationConfig.SSHKeys = &sshKeys
 		}
 
-		initializationUserDataFileID := initializationBlock[mkResourceVirtualEnvironmentVMInitializationUserDataFileID].(string)
+		password := initializationUserAccountBlock[mkResourceVirtualEnvironmentVMInitializationUserAccountPassword].(string)
 
-		if initializationUserDataFileID != "" {
-			initializationConfig.Files = &vms.CustomCloudInitFiles{
-				UserVolume: &initializationUserDataFileID,
-			}
+		if password != "" {
+			initializationConfig.Password = &password
 		}
 
-		initializationVendorDataFileID := initializationBlock[mkResourceVirtualEnvironmentVMInitializationVendorDataFileID].(string)
+		username := initializationUserAccountBlock[mkResourceVirtualEnvironmentVMInitializationUserAccountUsername].(string)
 
-		if initializationVendorDataFileID != "" {
-			if initializationConfig.Files == nil {
-				initializationConfig.Files = &vms.CustomCloudInitFiles{}
-			}
+		initializationConfig.Username = &username
+	}
 
-			initializationConfig.Files.VendorVolume = &initializationVendorDataFileID
+	initializationUserDataFileID := initializationBlock[mkResourceVirtualEnvironmentVMInitializationUserDataFileID].(string)
+
+	if initializationUserDataFileID != "" {
+		initializationConfig.Files = &vms.CustomCloudInitFiles{
+			UserVolume: &initializationUserDataFileID,
+		}
+	}
+
+	initializationVendorDataFileID := initializationBlock[mkResourceVirtualEnvironmentVMInitializationVendorDataFileID].(string)
+
+	if initializationVendorDataFileID != "" {
+		if initializationConfig.Files == nil {
+			initializationConfig.Files = &vms.CustomCloudInitFiles{}
 		}
 
-		initializationNetworkDataFileID := initializationBlock[mkResourceVirtualEnvironmentVMInitializationNetworkDataFileID].(string)
+		initializationConfig.Files.VendorVolume = &initializationVendorDataFileID
+	}
 
-		if initializationNetworkDataFileID != "" {
-			if initializationConfig.Files == nil {
-				initializationConfig.Files = &vms.CustomCloudInitFiles{}
-			}
+	initializationNetworkDataFileID := initializationBlock[mkResourceVirtualEnvironmentVMInitializationNetworkDataFileID].(string)
 
-			initializationConfig.Files.NetworkVolume = &initializationNetworkDataFileID
+	if initializationNetworkDataFileID != "" {
+		if initializationConfig.Files == nil {
+			initializationConfig.Files = &vms.CustomCloudInitFiles{}
 		}
 
-		//nolint:lll
-		initializationMetaDataFileID := initializationBlock[mkResourceVirtualEnvironmentVMInitializationMetaDataFileID].(string)
+		initializationConfig.Files.NetworkVolume = &initializationNetworkDataFileID
+	}
 
-		if initializationMetaDataFileID != "" {
-			if initializationConfig.Files == nil {
-				initializationConfig.Files = &vms.CustomCloudInitFiles{}
-			}
+	//nolint:lll
+	initializationMetaDataFileID := initializationBlock[mkResourceVirtualEnvironmentVMInitializationMetaDataFileID].(string)
 
-			initializationConfig.Files.MetaVolume = &initializationMetaDataFileID
+	if initializationMetaDataFileID != "" {
+		if initializationConfig.Files == nil {
+			initializationConfig.Files = &vms.CustomCloudInitFiles{}
 		}
 
-		initializationType := initializationBlock[mkResourceVirtualEnvironmentVMInitializationType].(string)
+		initializationConfig.Files.MetaVolume = &initializationMetaDataFileID
+	}
 
-		if initializationType != "" {
-			initializationConfig.Type = &initializationType
-		}
+	initializationType := initializationBlock[mkResourceVirtualEnvironmentVMInitializationType].(string)
+
+	if initializationType != "" {
+		initializationConfig.Type = &initializationType
 	}
 
 	return initializationConfig

